lib/course_planner_web/models/bulk_helper.ex

```diff
@@ -0,0 +1,78 @@
+defmodule CoursePlanner.BulkHelper do
+  @moduledoc """
+    Handle bulk creation specific logics
+  """
+  alias CoursePlanner.{Repo, Accounts.User}
+  alias Coherence.ControllerHelpers
+  alias Ecto.{DateTime, Multi}
+
+  def bulk_user_creation(csv_file_stream) do
+    csv_parsed_result = csv_parsing(csv_file_stream)
+
+    case validate_column_count(csv_parsed_result, 5) do
+      {:ok, parsed_params} ->
+        create_users_with_transaction(parsed_params)
+      {:validate_error, row_index} ->
+        {:error, "parsing_csv",
+          "Input data in row ##{row_index + 1} is not matching the column number.", ""}
+      {:error, :empty} ->
+        {:error, "parsing_csv", "Input can not be empty.", ""}
+      {:error, message} ->
+        {:error, "parsing_csv", message, ""}
+    end
+  end
+
+  defp csv_parsing(csv_file_stream) do
+    csv_file_stream
+    |> CSV.decode
+    |> Enum.reduce_while({:ok, []}, fn(parsed_row, {_out_result, out_value}) ->
+         case parsed_row do
+           {:ok, value} -> {:cont, {:ok, [value | out_value]}}
+           {:error, value} -> {:halt, {:error, value}}
+         end
+       end)
+  end
+
+  def validate_column_count({:error, parsed_params}, _column_count), do: {:error, parsed_params}
```

Comment:
you don't need to do this validation, the library will give you an error if one of the lines is bigger or smaller than the others, this includes empty lines

Reply:
so is it enough if I check the first line then ? to see if it has specifically 5 columns ?